packages/legend-graph/src/graphManager/action/changeDetection/TestObserverHelper.ts

```diff
@@ -0,0 +1,210 @@
+    });
+
+    return metamodel;
+  },
+);
+
+export const observe_AssertPass = skipObserved(
+  (metamodel: AssertPass): AssertPass => {
+    makeObservable(metamodel, {
+      id: observable,
+      hashCode: computed,
+    });
+
+    return metamodel;
+  },
+);
+
+export const observe_EqualToJsonAssertFail = skipObserved(
+  (metamodel: EqualToJsonAssertFail): EqualToJsonAssertFail => {
+    makeObservable(metamodel, {
+      id: observable,
+      message: observable,
+      actual: observable,
+      expected: observable,
+      hashCode: computed,
+    });
+
+    return metamodel;
+  },
+);
+
+function observe_AssertionStatus(metamodel: AssertionStatus): AssertionStatus {
+  if (metamodel instanceof EqualToJsonAssertFail) {
+    return observe_EqualToJsonAssertFail(metamodel);
+  } else if (metamodel instanceof AssertFail) {
+    return observe_AssertFail(metamodel);
+  } else if (metamodel instanceof AssertPass) {
+    return observe_AssertPass(metamodel);
+  }
+  return metamodel;
+}
+
+export const observe_EqualTo = skipObserved((metamodel: EqualTo): EqualTo => {
+  makeObservable(metamodel, {
+    id: observable,
+    expected: observable,
+    hashCode: computed,
+  });
+
+  return metamodel;
+});
+
+export const observe_EqualToJson = skipObserved(
+  (metamodel: EqualToJson): EqualToJson => {
+    makeObservable(metamodel, {
+      id: observable,
+      expected: observable,
+      hashCode: computed,
+    });
+
+    observe_ExternalFormatData(metamodel.expected);
+
+    return metamodel;
+  },
+);
+
+export const observe_TestError = skipObserved(
+  (metamodel: TestError): TestError => {
+    makeObservable(metamodel, {
+      testable: observable,
+      atomicTestId: observable,
+      error: observable,
+      hashCode: computed,
+    });
+
+    observe_AtomicTestId(metamodel.atomicTestId);
+
+    return metamodel;
+  },
+);
+
+export const observe_TestFailed = skipObserved(
+  (metamodel: TestFailed): TestFailed => {
+    makeObservable(metamodel, {
+      testable: observable,
+      atomicTestId: observable,
+      assertStatuses: observable,
+      hashCode: computed,
+    });
+
+    metamodel.assertStatuses.forEach(observe_AssertionStatus);
+    observe_AtomicTestId(metamodel.atomicTestId);
+
+    return metamodel;
+  },
+);
+
+export const observe_TestPassed = skipObserved(
+  (metamodel: TestPassed): TestPassed => {
+    makeObservable(metamodel, {
+      testable: observable,
+      atomicTestId: observable,
+      hashCode: computed,
+    });
+
+    observe_AtomicTestId(metamodel.atomicTestId);
+
+    return metamodel;
+  },
+);
+
+export const observe_TestResult = skipObserved(
+  (metamodel: TestResult): TestResult => {
+    makeObservable(metamodel, {
+      testable: observable,
+      atomicTestId: observable,
+      hashCode: computed,
+    });
+
+    observe_AtomicTestId(metamodel.atomicTestId);
+
+    return metamodel;
+  },
+);
+
+export function observe_AtomicTest(metamodel: AtomicTest): AtomicTest {
```

Comment:
Shouldn't this method take a context as well?

Reply:
We don't need that as of now. We will need it once we add stuff like AssertEqualTDS